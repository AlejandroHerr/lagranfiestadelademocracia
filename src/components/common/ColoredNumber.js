import React from 'react';
import PropTypes from 'prop-types';

const ColoredNumber = ({ children, negativeColor, positiveColor, renderComponent: RenderComponent, ...props }) => (
  <RenderComponent {...props} color={children < 0 ? negativeColor : positiveColor}>
    {children}
  </RenderComponent>
);

ColoredNumber.propTypes = {
  children: PropTypes.number.isRequired,
  negativeColor: PropTypes.string,
  positiveColor: PropTypes.string,
  renderComponent: PropTypes.func.isRequired,
};

ColoredNumber.defaultProps = {
  negativeColor: 'red',
  positiveColor: 'normal',
};

export default ColoredNumber;
