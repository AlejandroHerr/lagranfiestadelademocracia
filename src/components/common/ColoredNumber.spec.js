import React from 'react';
import renderer from 'react-test-renderer';

import ColoredNumber from './ColoredNumber';

describe('components/component/ColoredNumber', () => {
  it('renders correctly for positive numbers', () => {
    const tree = renderer
      .create(<ColoredNumber renderComponent={props => <div {...props} />}>{1}</ColoredNumber>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly for negative numbers', () => {
    const tree = renderer
      .create(<ColoredNumber renderComponent={props => <div {...props} />}>{-1}</ColoredNumber>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
