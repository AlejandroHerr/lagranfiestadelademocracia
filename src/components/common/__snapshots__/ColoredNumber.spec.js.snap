// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/component/ColoredNumber renders correctly for negative numbers 1`] = `
<div
  color="red"
>
  -1
</div>
`;

exports[`components/component/ColoredNumber renders correctly for positive numbers 1`] = `
<div
  color="normal"
>
  1
</div>
`;
